Tower

DLBPointon opened the pull request:
Adding sanger tower integration files.

@gq1 has suggested we default FULL in order to allow sanger testing to work. This PR does that and adds all the required files.

gq1 commented:
This is fine. The only problem I have is the profile names.

In all other pipelines, we use `test` and `test_full `as the profile name, instead yours `github_test `and `farm_test`

https://github.com/sanger-tol/insdcdownload/tree/dev/conf

In one of my following jobs, I only archive the `test_full `to results to S3,

gq1 commented:
I can't see any changes for the `main.nf`, how can you make the full workflow as the default workflow?

DLBPointon commented:
> This is fine. The only problem I have is the profile names.
> 
> In all other pipelines, we use `test` and `test_full `as the profile name, instead yours `github_test `and `farm_test`
> 
> https://github.com/sanger-tol/insdcdownload/tree/dev/conf
> 
> In one of my following jobs, I only archive the `test_full `to results to S3,

I'll rename it for consistency.

DLBPointon commented on .github/workflows/sanger_test.yml:
```diff
@@ -0,0 +1,29 @@
+            {
+              "outdir": "${{ secrets.TOWER_WORKDIR_PARENT }}/results/${{ github.repository }}/results-${{ github.sha }}",
+            }
+          profiles: farm_test,sanger,singularity,cleanup
```
I'll sort this out with the renaming i'm doing.

gq1 commented:
Are the profile `test` and `github_test `profile the same? Where are the test data? We can put the data into our S3 and then this profile should work anywhere.

DLBPointon commented:
They are the same, I have fixed that now.

What we want is small testing to be performed on GitHub, which we now have, and large genome testing on the farm. So this would be using `test_full` in the `sanger_test_full.config` wouldn't it? The current full test data is on lustre already in the `test-data` directory (we'll need to update that data to something bigger in the near future).

For test_full we can just use lustre path? If we use S3 for the full testing then we will need to package up the entire test directory because of how we have things set up, like we do with github testing at the minute.

I've deleted the sanger_test.config as I don't think we'll need the manual testing.

The local_github_testing was just so that I could test everything the whole github CI but locally. I can delete that now.

So for now, I just need to fix test_full for lustre paths to out large test set?

DLBPointon commented:
Working with Guoying yesterday, we have finalised everything needed for testing the pipeline on the Farm.

Waiting on final approval now.

gq1 commented:
I tried to run your pipeline on the farm and got one error here:

```
nextflow run treeval/ -profile test_full,sanger,singularity,cleanup --outdir test_results
N E X T F L O W  ~  version 23.04.1
ERROR ~ No such file or directory: Config file does not exist: /lustre/scratch123/tol/teams/tolit/users/gq2/git_test/treeval/conf/farm_test.config
```
The config file not updated here:
https://github.com/sanger-tol/treeval/blob/8e8f64b246e5abe862398f2dbe26a6fc29feaa83/nextflow.config#L156
